src/config.cc

```diff
@@ -5,200 +5,206 @@
+		} else if(str == "off") {
+			opt = celerity::detail::log_level::off;
+		} else if(str == "warn") {
+			opt = celerity::detail::log_level::warn;
+		} else if(str == "err") {
+			opt = celerity::detail::log_level::err;
+		} else if(str == "critical") {
+			opt = celerity::detail::log_level::critical;
+		} else if(str == "info") {
+			opt = celerity::detail::log_level::info;
+		} else {
+			throw parser_error{std::string("Unable to parse '") + std::string(str) + "'"};
+		}
+
+		return opt;
+	}
+};
+} // namespace env
 
 namespace celerity {
 namespace detail {
+
+	size_t config::parse_validate_graph_print_max_verts(const std::string_view str) {
+		const auto gmpv = env::default_parser<size_t>{}(str);
+		if(spdlog::should_log(celerity::detail::log_level::trace)) {
+			CELERITY_WARN("CELERITY_GRAPH_PRINT_MAX_VERTS: Graphs will only be printed for CELERITY_LOG_LEVEL=trace.");
+		}
+		CELERITY_WARN("CELERITY_GRAPH_PRINT_MAX_VERTS new value: {}.", gmpv);
+		return gmpv;
+	}
+
+	bool config::parse_validate_profile_kernel(const std::string_view str) {
+		const auto pk = env::default_parser<bool>{}(str);
+		CELERITY_WARN("CELERITY_PROFILE_KERNEL is {}.", pk ? "on" : "off");
+		return pk;
+	}
+
+	size_t config::parse_validate_dry_run_nodes(const std::string_view str) {
+		const size_t drn = env::default_parser<size_t>{}(str);
+		if(m_host_cfg.node_count != 1) throw std::runtime_error("In order to run with CELERITY_DRY_RUN_NODES a single MPI process/rank must be used.");
+		CELERITY_WARN("Performing a dry run with {} simulated nodes", drn);
+		return drn;
+	}
+
+	std::vector<size_t> config::parse_validate_devices(const std::string_view str) {
+		std::vector<size_t> devices;
+		const auto split_str = split(str, ' ');
+		// Delegate parsing of primitive types to the default_parser
+		for(size_t i = 0; i < split_str.size(); ++i) {
+			devices.push_back(env::default_parser<size_t>{}(split_str[i]));
+		}
+		if(devices.size() < 2)
+			throw env::validation_error{fmt::format(
+			    "Found {} IDs.\nExpected the following format: CELERITY_DEVICES=\"<platform_id> <first device_id> <second device_id> ... <nth device_id>\"",
+			    std::to_string(devices.size()))};
+
+		if(static_cast<long>(m_host_cfg.local_rank) > static_cast<long>(devices.size()) - 2) {
+			throw env::validation_error{fmt::format(
+			    "Process has local rank {}, but CELERITY_DEVICES only includes {} device(s)", m_host_cfg.local_rank, devices.empty() ? 0 : devices.size() - 1)};
+		}
+		if(static_cast<long>(devices.size()) - 1 > static_cast<long>(m_host_cfg.node_count)) {
+			throw env::validation_error{fmt::format("CELERITY_DEVICES contains {} device indices, but only {} worker processes were spawned on this host",
+			    devices.size() - 1, m_host_cfg.node_count)};
+		}
+
+		return devices;
+	}
+
+	bool config::parse_validate_force_wg(const std::string_view str) {
+		throw env::validation_error{fmt::format("Support for CELERITY_FORCE_WG has been removed with Celerity 0.3.0.")};
```

Comment:
There's no formatting going on here, so wrapping the string in `fmt::format` is pointless. Same goes for the function below.